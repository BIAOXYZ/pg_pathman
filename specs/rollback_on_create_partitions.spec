setup
{
	CREATE EXTENSION pg_pathman;
	CREATE TABLE range_rel(id serial primary key);
}

teardown
{
	SELECT drop_range_partitions('range_rel');
	DROP TABLE range_rel CASCADE;
	DROP EXTENSION pg_pathman;
}

session "s1"
step "begin" { BEGIN; }
step "rollback" { ROLLBACK; }
step "commit" { COMMIT; }
step "insert_data" { INSERT INTO range_rel SELECT generate_series(1, 10000); }
step "create_partitions" { SELECT create_range_partitions('range_rel', 'id', 1, 1000); }
step "show_me_what_you_got" { EXPLAIN (COSTS OFF) SELECT * FROM range_rel; }

permutation "begin" "insert_data" "create_partitions" "rollback" "show_me_what_you_got"
permutation "begin" "insert_data" "create_partitions" "commit" "show_me_what_you_got"
