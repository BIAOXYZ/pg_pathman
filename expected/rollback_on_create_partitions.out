Parsed test spec with 1 sessions

starting permutation: begin insert_data create_partitions rollback show_me_what_you_got
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions

10             
WARNING:  Removing 'range_rel' partitions from pg_pathman's cache
step rollback: ROLLBACK;
step show_me_what_you_got: EXPLAIN (COSTS OFF) SELECT * FROM range_rel;
QUERY PLAN     

Seq Scan on range_rel

starting permutation: begin insert_data create_partitions commit show_me_what_you_got
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions

10             
step commit: COMMIT;
step show_me_what_you_got: EXPLAIN (COSTS OFF) SELECT * FROM range_rel;
QUERY PLAN     

Append         
  ->  Seq Scan on range_rel_1
  ->  Seq Scan on range_rel_2
  ->  Seq Scan on range_rel_3
  ->  Seq Scan on range_rel_4
  ->  Seq Scan on range_rel_5
  ->  Seq Scan on range_rel_6
  ->  Seq Scan on range_rel_7
  ->  Seq Scan on range_rel_8
  ->  Seq Scan on range_rel_9
  ->  Seq Scan on range_rel_10
